### PR TITLE
ENG-5069 Sample application for working with multi file inputs/outputs

### PR DESCRIPTION
**Pull request by sebastian-quintero**
# Description

Adds a new community app: `knapsack-ortools-csv` with an example fo how to read from `input`, write to `output` and print stats to `stdout`.

Aligns some whitespacing, missing type hints and minor formatting in other apps.

### CODE REVIEW

File: knapsack-ortools-csv/main.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    print(message, file=sys.stderr)
+
+
+def read_input() -> dict[str, Any]:
+    """Reads the inputs from the .input dir and transforms into a single object."""
+
+    input_dir = ".input"
+
+    with open(f"{input_dir}/items.csv") as f:
+        reader = csv.DictReader(f, quoting=csv.QUOTE_NONNUMERIC)
```
Comment (merschformann) [inline knapsack-ortools-csv/main.py:123]: These options are the pain of CSV. :sweat_smile: 
I think for this example its a great default though. :+1:

**Review by merschformann — APPROVED** [review]
Thanks for improving the type hinting! :hugs: 

There is a failing test btw.

Looks good otherwise! :heart:

**Comment by nmisek** [comment]
noting here that we need to read from `input` and write to `output` without the `.`

File: knapsack-ortools-csv/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+items with a value and weight, our objective is to maximize the total value
+without exceeding the weight capacity of the knapsack.
+
+The `.input` folder defines the inputs of the problem:
+
+- `items.csv` contains the items to be considered in the knapsack problem.
+- `weight_capacity.csv` contains the weight capacity of the knapsack.
+
+The  `main.py` file implements a MIP knapsack solver. It reads from the
+`.input` folder, solves the problem, and writes the solution to an `.output`
```
Comment (dirkschumacher) [inline knapsack-ortools-csv/README.md]: `.input` or `input`?

**Review by dirkschumacher — CHANGES_REQUESTED** [review]

File: knapsack-ortools-csv/README.md (hunk 2 above)
Comment (sebastian-quintero) [reply knapsack-ortools-csv/README.md]: Thank you

**Review by dirkschumacher — APPROVED** [review]

**Pull request merged**